Model/SentryPerformance.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+declare(strict_types=1);
+
+namespace JustBetter\Sentry\Model;
+
+// phpcs:disable Magento2.Functions.DiscouragedFunction
+
+use JustBetter\Sentry\Helper\Data;
+use Laminas\Http\Response;
+use Magento\Framework\App\Http;
+use Magento\Framework\App\Request\Http as HttpRequest;
+use Magento\Framework\App\ResponseInterface;
+use Magento\Framework\App\State;
+use Magento\Framework\AppInterface;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\ObjectManagerInterface;
+use Sentry\SentrySdk;
+use Sentry\Tracing\SpanContext;
+use Sentry\Tracing\Transaction;
+use Sentry\Tracing\TransactionContext;
+use Sentry\Tracing\TransactionSource;
+use Throwable;
+
+use function Sentry\startTransaction;
+
+class SentryPerformance
+{
+    private ?Transaction $transaction = null;
+
+    public function __construct(
+        private HttpRequest $request,
+        private ObjectManagerInterface $objectManager,
+        private Data $helper
+    ) {
+    }
+
+    public function startTransaction(AppInterface $app): void
+    {
+        if (!$app instanceof Http) {
+            // actually, we only support profiling of http requests.
+            return;
+        }
+
+        $requestStartTime = $this->request->getServer('REQUEST_TIME_FLOAT', microtime(true));
+
+        $context = TransactionContext::fromHeaders(
+            $this->request->getHeader('sentry-trace') ?: '',
+            $this->request->getHeader('baggage') ?: ''
+        );
+
+        $requestPath = '/'.ltrim($this->request->getRequestUri(), '/');
+
+        $context->setName($requestPath);
+        $context->setSource(TransactionSource::url());
+        $context->setStartTimestamp($requestStartTime);
+
+        $context->setData([
+            'url'    => $requestPath,
+            'method' => strtoupper($this->request->getMethod()),
+        ]);
+
+        // Start the transaction
+        $transaction = startTransaction($context);
```

Comment:
Just to make sure. Starting a transaction here won't count towards your Sentry usage and limits right?
Since we'll be executing it every request

Reply:
no. it only counts if the report is sent to the gateway (at the latest moment)

But i did not tested it explicitly, because we do not have any limits.